src/commands/generationCommands.ts

```diff
@@ -38,6 +44,55 @@ export async function registerSelectTestFrameworkCommand(context: ExtensionConte
     }));
 }
 
+/**
+ * A command that server side can call to get a input value.
+ * Currently it's only used to get a Java qualified name, so the check is on the client side.
+ * @param context
+ */
+export async function registerAskForInputCommand(context: ExtensionContext): Promise<void> {
+    context.subscriptions.push(commands.registerCommand('_java.test.askClientForInput', async (prompt: string, value: string) => {
+        const ans: string | undefined = await window.showInputBox({
+            value,
+            prompt,
+            validateInput: checkJavaQualifiedName,
+        });
+        return ans;
+    }));
+}
+
+function checkJavaQualifiedName(value: string): string {
+    if (!value || !value.trim()) {
+        return 'Input cannot be empty.';
+    }
+
+    for (const part of value.split('.')) {
+        if (isKeyword(part)) {
+            return `Keyword '${part}' cannot be used.`;
+        }
+
+        if (!isJavaIdentifier(part)) {
+            return `Invalid Java qualified name.`;
+        }
+    }
+
+    return '';
+}
+
+const keywords: Set<string> = new Set([
```

Comment:
where do you find the keywords?

Reply:
Good point. I copied the code from the project manager.

There is a official list(JAVA 8) here: https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-Keyword

I think I should update to this one.

Reply:
My point is, also to place the link to source in comments.

Reply:
You should refer to the latest Java 16 specs.
https://docs.oracle.com/javase/specs/jls/se16/html/jls-3.html#jls-Keyword

_(underscore) is disallowed as identifier.